### ISSUE TITLE
CSS Naming Conventions

### ISSUE THREAD

**Issue by wellington-damasio** [wellington-damasio/github-blog_issues#3]
The way you name classes in your CSS stylesheet can either improve the maintenance of your codebase or make everyone wish they were alone with sharks in the middle of the ocean rather than maintaining your dirty CSS (no offense 🙃).

Every one of us wrote bad code, and bad CSS class names one day. But starting now, you're not going to be on the list of people that do this anymore (and I hope aren't already).

How to write good CSS class names
There are two ways I like to write CSS class names (if you know any better way to write class names in CSS feel free to do so and please tell me about that, I'm always open to new ways of doing things).

1. Parent/context classes
Parent/context is a name I came up with to describe the way I think about class names in CSS.

It is a way to style elements inside a component without using too many CSS selectors like .user-card > div > div p or having to write things repeatedly like user-card-name user-card-role user-card-button.

In the user card example, I used the .user-card class as a parent/context for other classes that were inside the user card like .name .role button, etc. So if there were any other .name .role button styles in our CSS they wouldn't be affected by the changes I made to user card styles.

2. [BEM](https://getbem.com/naming/)
BEM is a CSS naming convention and it stands for Block Element Modifier.

BEM is used in a lot of different projects and by lots of developers so I recommend checking it out.

Basically, classes are considered blocks, and classes inside these blocks are elements.

Modifiers are classes that are used for dynamic styling (for example a button that changes background colors through the application).

To target an element inside a block in [BEM CSS](https://getbem.com/naming/), for (example our .user-card .name). We would have to write it the following way: .user-card__name

This can seem very repetitive at first but when you're working with a team, this is a very intuitive way to standardize class names in CSS in a way that everyone understands.

I used the [BEM](https://getbem.com/naming/) naming convention in this mini-project about CSS Grid.

.grid (block)

.grid__item (element)

btn--remove, btn--change (modifiers)
